src/software/jetson_nano/services/motor.cpp

```diff
@@ -102,54 +111,223 @@ MotorService::MotorService(const RobotConstants_t& robot_constants,
+                     << "The driver becomes disabled until flag becomes cleared.";
+    }
+
+    if (gstat_bitset[5])
+    {
+        LOG(WARNING) << "s2gu: Short to GND detected on phase U."
+                     << "The driver becomes disabled until flag becomes cleared.";
+    }
 
-            // TODO (#2456) We can only spin 1 motor right now, figure out how
-            // to spin the rest and update the following code
+    if (gstat_bitset[6])
+    {
+        LOG(WARNING) << "s2vsu: Short to VS detected on phase U."
+                     << "The driver becomes disabled until flag becomes cleared.";
+    }
+
+    if (gstat_bitset[8])
+    {
+        LOG(WARNING) << "shortdet_v: V short counter has triggered at least once.";
+    }
+
+    if (gstat_bitset[9])
+    {
+        LOG(WARNING) << "s2gv: Short to GND detected on phase V."
+                     << "The driver becomes disabled until flag becomes cleared.";
+    }
+
+    if (gstat_bitset[10])
+    {
+        LOG(WARNING) << "s2vsv: Short to VS detected on phase V."
+                     << "The driver becomes disabled until flag becomes cleared.";
+    }
+
+    if (gstat_bitset[12])
+    {
+        LOG(WARNING) << "shortdet_w: short counter has triggered at least once.";
+    }
+
+    if (gstat_bitset[13])
+    {
+        LOG(WARNING) << "s2gw: Short to GND detected on phase W."
+                     << "The driver becomes disabled until flag becomes cleared.";
+    }
+
+    if (gstat_bitset[14])
+    {
+        LOG(WARNING) << "s2vsw: Short to VS detected on phase W."
+                     << "The driver becomes disabled until flag becomes cleared.";
+    }
+
+    return !gstat_bitset.any();
+}
+
+
+TbotsProto::MotorStatus MotorService::poll(const TbotsProto::MotorControl& motor)
+{
+    CHECK(encoder_calibrated_[FRONT_LEFT_MOTOR_CHIP_SELECT] &&
+          encoder_calibrated_[FRONT_RIGHT_MOTOR_CHIP_SELECT] &&
+          encoder_calibrated_[BACK_LEFT_MOTOR_CHIP_SELECT] &&
+          encoder_calibrated_[BACK_RIGHT_MOTOR_CHIP_SELECT])
+        << "Running without encoder calibration can cause serious harm, exiting";
+
+    int front_right_rpm = tmc4671_getActualVelocity(FRONT_RIGHT_MOTOR_CHIP_SELECT);
+    int front_left_rpm  = tmc4671_getActualVelocity(FRONT_LEFT_MOTOR_CHIP_SELECT);
+    int back_right_rpm  = tmc4671_getActualVelocity(BACK_RIGHT_MOTOR_CHIP_SELECT);
+    int back_left_rpm   = tmc4671_getActualVelocity(BACK_LEFT_MOTOR_CHIP_SELECT);
+
+    WheelSpace_t current_wheel_speeds = {
+        static_cast<double>(front_left_rpm), static_cast<double>(front_right_rpm),
+        static_cast<double>(back_left_rpm), static_cast<double>(back_right_rpm)};
+
+    TbotsProto::MotorStatus motor_status;
+    motor_status.mutable_front_right()->set_wheel_rpm(front_right_rpm);
+    motor_status.mutable_front_left()->set_wheel_rpm(front_left_rpm);
+    motor_status.mutable_back_left()->set_wheel_rpm(back_left_rpm);
+    motor_status.mutable_back_right()->set_wheel_rpm(back_right_rpm);
```

Comment:
generally seems too easy to introduce bugs when we have to match front_right and front_right, etc.

Reply:
not sure how this can be improved, it has to be set somewhere